From eb32e0f596a4b4d0d8480b0bc8da88bc25a0c691 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Fri, 20 Jan 2023 16:57:17 +0100
Subject: Do not load updater GN if it is disabled

Recent changes to make chrome updater depend on systemd broke build
on Linux systems without systemd, even if enable_updater is set to
false.

This change allows systems without systemd, but without building
updater (as in some Yocto based builds as meta-chromium) to build
successfully.

Bug: 1409085
Change-Id: Icb763fa728da7e6c2ec14133973bc762ebbc99d7
---
 BUILD.gn | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/BUILD.gn b/BUILD.gn
index dd884bc226dbf..285eeb042dc7c 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -153,10 +153,11 @@ group("gn_all") {
   }
 
   if (!is_castos) {
-    deps += [
-      "//chrome/installer",
-      "//chrome/updater",
-    ]
+    deps += [ "//chrome/installer" ]
+  }
+
+  if (enable_updater) {
+    deps += [ "//chrome/updater" ]
   }
 
   if (enable_ppapi) {
-- 
2.37.2

